### PR TITLE
Add Streptococcus pneumoniae subworkflow

### PR DESCRIPTION
**Pull request by kapsakcj**
This PR adds:

* a subworkflow to `wf_merlin_magic.wdl` which runs the following tools on any sample identified as `Streptococcus pneumoniae` by GAMBIT:
  * [pbptyper](https://github.com/rpetit3/pbptyper) - in silico Penicillin Binding Protein (PBP) typing
  * [seroBA](https://github.com/sanger-pathogens/seroba) - S. pneumoniae serotyping
  * [PopPUNK](https://github.com/bacpop/PopPUNK) + [Global Pneumococcal Sequencing (GPS) database v6](https://www.pneumogen.net/gps/training_command_line.html) for GPS Cluster assignment

Setting as a draft for now, since I believe that TheiaProk_Illumina_SE needs a few more things added (seroBA outputs? other outputs?) and subsequent testing in Terra

### CODE REVIEW

**Comment by kapsakcj** [comment]
Brought in the changes from PR #137 & main into this PR (see last commit ^), hopefully will resolve the `pytest` errors, but we can iron them out if there are any further

**Comment by kapsakcj** [comment]
New outputs from these 3 tasks. We could eventually use this in future documentation?

| output type | output name (column name in Terra) | short description |
| :------: | ------ |  ------ |
| String | pbptyper_predicted_1A_2B_2X | predicted PBP type in `#:#:#` format, ex: `23:0:2`. Corresponds with (1A:2B:2X genes)   |
| File | pbptyper_pbptype_predicted_tsv | tab-delimited file with the predicted PBP type |
| String? | pbptyper_version | self-explanatory |
| String? | poppunk_gps_cluster | Predicted GPS Cluster number |
| File? | poppunk_gps_external_cluster_csv | Simple TSV containing the GPS Cluster number |
| String? | poppunk_version | self-explanatory |
| String? | seroba_version | self-explanatory |
| String? | seroba_serotype | Predicted S. pneumoniae serotype |
| String? | seroba_contamination | Will print `contamination` if contamination is detected - at least one heterozygous SNP detected in read data with at least 10% of the mapped reads supporting the position |
| String? | seroba_ariba_serotype | Predicted S. pneumoniae serotype, but by ariba |
| String? | seroba_ariba_identity | % identity of sample's ariba assembly vs. ref sequence |
| File? | seroba_details | summary text file (NOT a TSV) with the above info |

**Comment by kapsakcj** [comment]
Getting this error with pytest & miniwdl now `VERBOSE BadRequestException: 400 Bucket is a requester pays bucket but no user project provided.` due to the GPS database files.

Honestly the DB files might be too large to test with on GHActions.

I wonder if we could disable the downloading of these database files? They aren't necessary for testing the whole TheiaProk_Illumina_PE workflow, only for the S. pneumoniae-specific sub-workflow

**Comment by kapsakcj** [comment]
I'm doing 2 final big tests. 100 Strep pneumoniae isolates through PE and SE workflows, call caching off so everything is re-run, and lastly testing export_taxon_table functionality to 2 separate Terra data tables.

SE workflow - https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/curtis_sandbox/job_history/4ce1840f-e4aa-4e62-a02c-ca44a81e9172

PE workflow - https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/curtis_sandbox/job_history/5f39d902-dd35-4a6f-8097-45c4a425723e

I expect 5 failures per workflow, each at the trimmomatic step. Something is wrong with the FASTQ files (no quality scores included?). I just got unlucky with those 5 ERR accessions.

**Comment by kapsakcj** [comment]
Looks like the export_taxon_tables task needs tweaking. Workflows ran as expected, except export taxon tables. Likely just need to fix table generation code

```
Sample ERR1065126 identified as Streptococcus pneumoniae. As per user-defined taxon tables, Streptococcus_pneumoniae samples will be exported to the strep_pneumoniae_specimen terra data table
0%| | 0/1 [00:00<?, ?it/s] 100%|██████████| 1/1 [00:00<00:00, 1.80it/s] 100%|██████████| 1/1 [00:00<00:00, 1.80it/s]
{
"causes": [],
"message": "TSV parsing error in line 0: wrong number of fields",
"source": "FireCloud",
"stackTrace": [],
"statusCode": 400,
"timestamp": 1661450641695
}
```

**Comment by kapsakcj** [comment]
export_taxon_tables task works as expected. The user just has to be careful to save their input taxon_table TSV with linux file endings (LF) and NOT windows (CRLF).

**Comment by rpetit3** [comment]
Going to have to figure out how to deal with `gs://theiagen-public-files-rp/terra/theiaprok-files/GPS_v6/GPS_v6_fit.npz` for the CI

```
2022-08-24 22:18:44.882 wdl.w:theiaprok_illumina_pe INFO start download :: uri: "gs://theiagen-public-files-rp/terra/theiaprok-files/GPS_v6/GPS_v6_fit.npz"
2022-08-24 22:18:44.882 wdl.w:theiaprok_illumina_pe NOTICE aborting workflow
2022-08-24 22:18:44.887 wdl.w:theiaprok_illumina_pe ERROR workflow theiaprok_illumina_pe (./workflows/wf_theiaprok_illumina_pe.wdl Ln 20 Col 1) failed :: dir: "/home/runner/pytest_workflow_q2i86in2/theiaprok_illumina_pe_miniwdl/miniwdl_run", error: "DownloadFailed", message: "unable to download gs://theiagen-public-files-rp/terra/theiaprok-files/GPS_v6/GPS_v6_unword_clusters.csv"

```

**Comment by kapsakcj** [comment]
Preferably we could skip the Poppunk/GPS task that requires those database files. The database alone is like 22GB so not worth trying to test it with every round of CI. That's something we can test in Terra before releases and such, but not good to do for every push/PR/etc.

I wonder if there's a way via `miniwdl` to skip a certain portion of a workflow

**Comment by rpetit3** [comment]
Hmmm, there might be. Another option would be to add a parameter to skip this step in the pipeline

**Comment by sage-wright** [comment]
all checks passed!

File: tests/inputs/empty-for-test.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+This file is empty for test purposes.
```
Comment (kevinlibuit) [inline tests/inputs/empty-for-test.txt:1]: @sage-wright still needed?

Comment (sage-wright) [reply tests/inputs/empty-for-test.txt:1]: Yes! Keep this file.

Comment (sage-wright) [reply tests/inputs/empty-for-test.txt:1]: This is the file that we're passing in as input for the poppunk task so it doesn't try to find the `gs://` files

**Review by kevinlibuit — APPROVED** [review]

**Pull request merged**